### PR TITLE
Addiction Rework

### PR DESCRIPTION
**Pull request by Fox-McCloud**
This is very heavily inspired by Goon's addiction system.

Addictions are now based on probability on INGESTing a reagent rather than based on surpassing a particular threshold. 

Addictions still progress in stages, but stage advancement is based on probability every time a reagent tick. All addictive things now have the same impacts--at the final stage, you'll start gagging and falling on the floor; it's very debilitating. After 8 minutes has passed, you can start rolling for a chance to be relieved of your addiction to a particular chemical.

If you take more of a reagent you're addicted to it will reset your addiction stage back to 1, but it will also reset the 8 minute timer you must pass for you to be free of an addiction.

This should make addictions more realistic, in terms of behavior, while not impacting game balance horrendously, while this also differentiates addictions better from overdosing.

Related
- Health scanners will tell what stage of addiction a person is at for what they're addicted to
- Cigs don't proc INGEST (too much spam, otherwise)
- the follow reagents are addictive: meth, crank, krokodil, bath salts, space drugs, perfluoradecalin, omnizine, coffee, nicotine, fliptonium, ultra lube, surge, fake cheese, weird cheese, ephedrine, diphenhydramine, teporone, and morphine.
 - Sleepers help you recover from addiction faster

:cl: Fox McCloud
rscadd: Reworks addictions to bettter differentiate them betweeen overdosing and make them more realistic.
tweak: changes which chems are addictive, currently, the follow reagents are now addidctive: meth, crank, krokodil, bath salts, space drugs, perfluoradecalin, omnizine, coffee, nicotine, fliptonium, ultra lube, surge, fake cheese, weird cheese, ephedrine, diphenhydramine, teporone, and morphine
rscadd: sleepers help you recover from addiction faster
/:cl:

### CODE REVIEW

**Comment by pinatacolada** [comment]
no way to treat addictions asides RNG?

**Comment by Fox-McCloud** [comment]
>no way to treat addictions asides RNG?

You can take more of the chemical to reset your addiction stage back to 1, but that also resets the 8 minute timer, too.

File: code/modules/reagents/newchem/medicine.dm [hunk 1]
```diff
@@ -336,6 +337,7 @@ datum/reagent/perfluorodecalin
 	reagent_state = LIQUID
 	color = "#C8A5DC"
 	metabolization_rate = 0.2
+	addiction_chance = 20
```
Comment (Crazylemon64) [inline code/modules/reagents/newchem/medicine.dm]: Hmm, this should offer a good reason to make salbut in place of perfluoro, now...

**Comment by TullyBurnalot** [comment]
Oh god, Perfluoro is addictive?

*scream

File: code/modules/reagents/oldchem/reagents/drink/reagents_alcohol.dm [hunk 2]
```diff
@@ -17,6 +17,7 @@
 	var/vomit_start = 180	//amount absorbed after which mob starts vomitting
 	var/blur_start = 260	//amount absorbed after which mob starts getting blurred vision
 	var/pass_out = 325	//amount absorbed after which mob starts passing out
+	addiction_chance = 4 //chance to become alcoholic
```
Comment (Crazylemon64) [inline code/modules/reagents/oldchem/reagents/drink/reagents_alcohol.dm]: Hmm, this seems like alcohol will have a 33% chance to cause an addiction, if the person drinks 2 50-u glasses at 10u per sip.

File: code/modules/reagents/newchem/drugs.dm [hunk 3]
```diff
@@ -532,7 +413,7 @@
 
 	process_flags = SYNTHETIC
 	overdose_threshold = 20
-	addiction_threshold = 10
+	addiction_chance = 60
 	metabolization_rate = 0.6
```
Comment (Crazylemon64) [inline code/modules/reagents/newchem/drugs.dm]: These addictions would be gained if you drank from glasses, right?

File: code/modules/reagents/newchem/medicine.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/medicine.dm]: Yup.

File: code/modules/reagents/newchem/drugs.dm (hunk 3 above)
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/drugs.dm]: Yeah, I'll likely need to tweak that a bit.

**Comment by FalseIncarnate** [comment]
Can't say I like the idea of addictions being RNG-based... especially for the first unit you take. If it were perhaps a chance that scaled with the amount or something, that'd be different. As it stands in this change, you can get screwed over by even 0.1 of a reagent having been slipped into your drink at the bar. Obviously that wouldn't trigger beyond a 1st or 2nd stage addiction, but the RNG of it is just unsavory.

**Comment by ProperPants** [comment]
>If you take more of a reagent you're addicted to it will reset your addiction stage back to 1, but it will also reset the 8 minute timer you must pass for you to be free of an addiction.

I don't understand this. Is stage 1 the least extreme or most extreme? I'm interpreting this as stage 1 being the least extreme, but if this is the case, continuously taking the reagent would be easier than not. What does it matter if the 8 minute timer resets, if you are at the lowest stage of addiction. The lowest stage isn't the debilitating one.

Of course stage 1 could be the worst, but your phrasing makes me think otherwise. The timer reset is a bad thing, so stage 1 must be a good thing.

One more note, does it reset you to stage 1, only to go up and up through the stages until the final stage? But at that point, if you're laying on the floor, how can you take more reagent to reset? Someone will drag you to medical, where they will see your addiction, cure it, and then the timer and the reset with more reagent don't matter because you aren't addicted anymore.

I must be missing something, but I really don't see what. Could someone explain this to me?

**Comment by FlattestGuitar** [comment]
I don't like the idea of getting addicted to nicotine or alcohol. Seems silly to me that something that's mostly used for RP is being nerfed.

**Comment by Fox-McCloud** [comment]
Alright.

You take a reagent that's addictive, and you become addicted to that reagent.

You start at stage 1---every so often, the stage will increase by 1. When you hit stage 5, you will start falling down.

If, at any point, you take the reagent you're addicted to, the addiction stage will reset to 1.

When you become addicted to a reagent, there's a timer that is set for 8 minutes. Once that 8 minutes are up, your addiction will end. That said, if you take the reagent that you're addicted to, that 8 minute timer will be reset.

>I don't like the idea of getting addicted to nicotine or alcohol. Seems silly to me that something that's mostly used for RP is being nerfed.

@FlattestGuitar 

Nicotine is *already* addictive, not to mention a combat chem---the chance for alcohol addiction is very small (4%)--not to mention that, well, alcoholism addiction is a thing; it's called alcoholism.

**Comment by Iamgoofball** [comment]
@FlattestGuitar Nicotine has stun reduction.

**Comment by ProperPants** [comment]
@Fox-McCloud 
Oh! I didn't realize that once you _do_ hit the addiction proc (if proc is even the right word in this situation), the stages progress with time. I assume you hit stage 5 before the 8 minutes are up. This makes total sense then, and it does seem to parallel actual addiction.
:+1: :+1:

**Comment by Fox-McCloud** [comment]
@ProperPants 

Oh yeah, you're going to hit stage 5 well before the 8 minutes are up; you're going to have to deal with the debilitating effects of addiction for a bit before the addiction wears off.

**Comment by FlattestGuitar** [comment]
If there are any people who actually use nicotine to get the edge in combat, they can just use patches to bypass the threat of addiction. Most people don't use them for the stun reduction, as nicotine is pretty weak when compared to any other combat stim.

@ProperPants, both alcohol and nicotine are hard drugs, meaning the addict on a physical level, as well as psychological. My issue here is that it would take A LOT more time to get addicted to either of those than meth or krokodil.

**Comment by Keilden** [comment]
Sugar should also be addictive, Sugar can have a heroin like addiction.

**Comment by Fox-McCloud** [comment]
- removed ethanol's addictiveness
- ensured that synths can't get addicted to organic drugs and visa-versa
- sleepers help you get over addiction faster.

**Pull request merged**